partiql-ast/src/main/resources/partiql_ast.ion

```diff
@@ -0,0 +1,624 @@
+  },
+
+  // The `like` SQL special form
+  like::{
+    value:    expr,
+    pattern:  expr,
+    escape:   optional::expr,
+  },
+
+  // The `between` SQL special form
+  between::{
+    value:  expr,
+    from:   expr,
+    to:     expr,
+  },
+
+  // The `in` SQL special form
+  in_collection::{
+    lhs: expr,
+    rhs: expr,
+  },
+
+  // The `is` SQL special form
+  is_type::{
+    value:  expr,
+    type:   '.type',
+  },
+
+  // The simple and searched `case` switch SQL special form F261-01, F261-02
+  switch::{
+    expr:     optional::expr,
+    branches: list::[branch],
+    default:  optional::expr,
+    _: [
+      branch::{
+        condition:  expr,
+        expr:       expr,
+      },
+    ],
+  },
+
+  // The `coalesce` SQL special form F261-04
+  coalesce::{
+    args: list::[expr],
+  },
+
+  // The `nullif` SQL special form F261-03
+  null_if::{
+    expr0: expr,
+    expr1: expr,
+  },
+
+  // The `cast` SQL special form F201
+  cast::{
+    value:    expr,
+    as_type:  '.type',
+  },
+
+  // The `can_cast` PartiQL special form
+  can_cast::{
+    value:    expr,
+    as_type:  '.type',
+  },
+
+  // The `can_lossless_cast` PartiQL special form
+  can_lossless_cast::{
+    value:    expr,
+    as_type:  '.type',
+  },
+
+  // The [INNER|OUTER] [UNION|INTERSECT|EXCEPT] PartiQL multiset special forms
+  set::{
+    op:         [ UNION, INTERSECT, EXCEPT ],
+    quantifier: set_quantifier,
+    outer:      bool,
+    lhs:        expr,
+    rhs:        expr,
+  },
+
+  // The PartiQL `sfw` query expression
+  s_f_w::{
+    select:   select,
+    from:     from,
+    let:      optional::let,
+    where:    optional::expr,
+    group_by: optional::group_by,
+    having:   optional::expr,
+    order_by: optional::order_by,
+    limit:    optional::expr,
+    offset:   optional::expr,
+  },
+
+  // GPML ( <expr> MATCH <pattern> )
+  match::{
+    expr:     expr,
+    pattern:  graph_match,
+  },
+
+  // WINDOW
+  window::{
+    function: string,
+    over:     over,
+    args:     list::[expr],
+  },
+]
+
+// PartiQL SELECT Clause Variants — https://partiql.org/dql/select.html
+select::[
+
+  // SELECT [ALL|DISTINCT] *
+  star::{
+    quantifier: set_quantifier,
+  },
+
+  // SELECT [ALL|DISTINCT] <select-list>
+  project::{
+    quantifier: set_quantifier,
+    items: list::[item],
+    _: [
+      item::[
+        all::{ expr: expr },                            // <expr>.*
+        var::{ expr: expr, as_alias: optional::string } // <expr> [as <identifier>]
+      ],
+    ],
+  },
+
+  // PIVOT <expr> AT <expr>
+  pivot::{
+    value:  expr,
+    key:    expr,
+  },
+
+  // SELECT [ALL|DISTINCT] VALUE <expr>
+  value::{
+    quantifier: set_quantifier,
+    constructor: expr,
+  },
+]
+
+// PartiQL FROM Clause Variants — https://partiql.org/dql/from.html
+from::[
+
+  // FROM [UNPIVOT] <expr> [AS <identifier>] [AT <identifier>] [BY <identifier>]
+  collection::{
+    expr:     expr,
+    unpivot:  optional::bool,
```

Comment:
I'm avoid "scan" as this is the AST. UNPIVOT is kind of a flag in the AST since it's an optional keyword. Not much different than say `SELECT [ALL|DISTINCT]` which is a flag indicating different semantics.

I'll split the FROM to model UNPIVOT similar to `partiql.ion`

Reply:
> If the motivation is in avoiding the repetition of the alias fields, maybe bundle them into an AST node? 

This is what #242 is doing. I'll add that here

Reply:
This is still a "flag" I suppose